test/testinput/bufr_ncep_adpsfc.yaml

```diff
@@ -0,0 +1,277 @@
+# (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+# #
+# # This software is licensed under the terms of the Apache Licence Version 2.0
+# # which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+observations:
+  - obs space:
+      name: bufr
+
+      obsdatain: "testinput/gdas.t12z.adpsfc_thinned.tm00.bufr_d"
+
+      exports:
+        variables:
+          timestamp:
+            datetime:
+              year: "*/YEAR[1]"
+              month: "*/MNTH[1]"
+              day: "*/DAYS[1]"
+              hour: "*/HOUR[1]"
+              minute: "*/MINU[1]"
+          altimeterSettingQnh:
+            query: "*/ALSE"
+          cloudAmount:
+            query: "[*/CLDSQ1/CLAM, */MTRCLD/CLAM, NC000100/CLAM[1], NC000101/CLAM[1], NC000102/CLAM[1]]"
+          latitude:
+            query: "[*/CLAT, */CLATH]"
+          longitude:
+            query: "[*/CLON, */CLONH]"
+          heightAboveSurfaceOfBaseOfLowestCloudSeen:
+            query: "*/HBLCS"
+          heightOfBaseOfCloud:
+            query: "[*/CLDSQ1/HOCB, */MTRCLD/HOCB, NC000100/HOCB, NC000101/HOCB, NC000102/HOCB]"
+          horizontalVisibility:
+            query: "*/HOVI"
+          heightOfStationGroundAboveMeanSeaLevel:
+            query: "*/HSMSL"
+          minimumTemperatureAtHeightAndOverPeriodSpecified:
+            query: "[*/TMPSQ3/MITM, */MTTPSQ/MITM, NC000100/BSYEXTM/MITM[1], NC000101/BSYEXTM/MITM[1], NC000102/BSYEXTM/MITM[1]]"
+          maximumWindGustSpeed:
+            query: "[*/WNDSQ2/MXGS, */BSYWND2/MXGS]"
+          maximumTemperatureAtHeightAndOverPeriodSpecified:
+            query: "[*/TMPSQ3/MXTM, */MTTPSQ/MXTM, */BSYEXTM/MXTM]"
+          pressureReducedToMeanSeaLevel:
+            query: "*/PMSL"
+          pressure:
+            query: "*/PRES"
+          presentWeather:
+            query: "[*/PPWSQ1/PRWE, */MTRPRW/PRWE, */PWEATHER/PRWE]"
+          airTemperatureQM:
+            query: "*/QMAT"
+          moistureQM:
+            query: "*/QMDD"
+          pressureQM:
+            query: "*/QMPR"
+          reportIdentifier:
+            query: "*/RPID"
+          stationElevation:
+            query: "*/SELV"
+          waterTemperature:
+            query: "*/SST1"
+          airTemperature:
+            query: "*/TMDB"
+          dewpointTemperature:
+            query: "*/TMDP"
+          cloudCoverTotal:
+            query: "*/TOCC"
+          verticalSignificanceSurfaceObservations1:
+            query: "[*/CLDSQ1/VSSO, */MTRCLD/VSSO, NC000100/VSSO[1], NC000101/VSSO[1], NC000102/VSSO[1]]"
+          verticalSignificanceSurfaceObservations2:
+            query: "[NC000100/VSSO[2], NC000101/VSSO[2], NC000102/VSSO[2]]"
+          verticalSignificanceSurfaceObservationsSupplemental:
+            query: "*/BSYSCLD/VSSO"
+          verticalSignificanceSurfaceObservationsBelowStation:
+            query: "*/BSYBCLD/VSSO"
```

Comment:
```
| NC000102 | MOBIDENT  WIGOSID  RPID  PRESDATA  QMPR  TEMHUMDA  QMAT  QMDD     |
| NC000102 | <BSYEXTM>  VISBSEQN  BSYWND1  QMWN  {BSYWND2}  TPMI  BSYPCP1      |
| NC000102 | {BSYPCP2}  HSALG  HSAWS  GENCLOUD  {BSYSCLD}  {BSYBCLD}           |
| NC000102 | <DIRCLDFT>  VSSO  <PWEATHER>  <STGDSNDM>  GMITH  BID  RCPTIM      |
| NC000102 | CORN                                                                           
```
```
| GENCLOUD | TOCC  VSSO  CLAM  HOCB  CLTP  CLTP  CLTP                          |

```
GSI is using UFBINT to retrieve VSSO like the following:

```
data cldseqstr /'VSSO CLAM HOCB'/   ! vertical significance, cloud amount and cloud base height
real (r_double), dimension(3,10):: cldseq
call ufbint (lunin, cldseq,3, 10, cldseqlevs, cldseqstr)
```

So, for VSSO, only the root level data is retrieved from GENCLOUD. We do not need to include VSSO from 
 {BSYSCLD}  {BSYBCLD} .

Here is my suggestion - change lines 68-75 to the following:

```
verticalSignificanceSurfaceObservations:
      query: "[NC000100/VSSO[1], NC000101/VSSO[1], NC000102/VSSO[1], */CLDSQ1/VSSO, */MTRCLD/VSSO]"
```

Reply:
@rmclaren Are you suggesting the following?
```
verticalSignificanceSurfaceObservations:
      query: "[*/VSSO[2], */CLDSQ1/VSSO, */MTRCLD/VSSO]"
```

We trying to get the VSSO from GENCLOUD

Reply:
@rmclaren Actually, we have a question about VSSO.  I will set up a google meet with you tomorrow.

Reply:
So the version of gdas.t12z.adpsfc_thinned.tm00.bufr_d that is checked in only seems to have data for NC000000 and NC000001...

Reply:
@nicholasesposito We need the test data set with all 7 subsets.  
One way to do this is to thin each subset first and then concatenate them together (on HERA, just use "cat" to add all 7 thinned subsets into one thinned file)

Reply:
@rmclaren I added the issue about VSSO in the spreadsheet.  
Here is the summary:
(1) issue related to print_query for VSSO
For subsets NC000100, NC000101, and NC000102:
The print_query should list vsso as following:
1d   NC000100   VSSO[1]
1d   NC000100   VSSO[2]

Instead, the following were found — VSSO indexes are missing
1d   NC000100   VSSO
1d   NC000100   VSSO

(2) query problem with VSSO  (please see related BUFR table attached below)

This does not work: (but it should) 
```
verticalSignificanceSurfaceObservations:
   query: "[*/VSSO[1], */CLDSQ1/VSSO, */MTRCLD/VSSO]"
```
The following two ways worked:
```
verticalSignificanceSurfaceObservations:
   query: "[NC000100/VSSO[1], NC000101/VSSO[1], NC000102/VSSO[1], */CLDSQ1/VSSO, */MTRCLD/VSSO]"
```
```
verticalSignificanceSurfaceObservations:
   query: "[*/CLDSQ1/VSSO, */MTRCLD/VSSO, */VSSO[1]]"
```

```
|------------------------------------------------------------------------------|
| MNEMONIC | SEQUENCE                                                          |
|----------|-------------------------------------------------------------------|
| NC000000 | YYMMDD  HHMM  RPID  WMOB  WMOS  LALOLV  CORN  RSRD  EXPRSRD  BID  |
| NC000000 | RCPTIM  RPSEC1  WNDSQ1  TMPSQ1  PRSSQ1  <PRSSQ2>  <PRSSQ3>        |
| NC000000 | PCPSQ1  TOCC  HBLCS  {CLDSQ1}  <PPWSQ1>  <WAVSQ1>  <WAVSQ2>       |
| NC000000 | {WAVSQ3}  <SNWSQ1>  <ICESQ1>  <GRDSQ1>  <CTMPS5>  <RPSEC3>        |
| NC000000 | {RAWRPT}                                                          |
|          |                                                                   |
| NC000001 | YYMMDD  HHMM  RPID  WMOB  WMOS  LALOLV  CORN  BID  RCPTIM  RPSEC1 |
| NC000001 | WNDSQ1  TMPSQ1  PRSSQ1  <PRSSQ2>  <PRSSQ3>  PCPSQ1  TOCC  HBLCS   |
| NC000001 | {CLDSQ1}  <PPWSQ1>  <WAVSQ1>  <WAVSQ2>  {WAVSQ3}  <SNWSQ1>        |
| NC000001 | <ICESQ1>  <GRDSQ1>  <CTMPS5>  <RPSEC3>  {RAWRPT}                  |
|          |                                                                   |
| NC000002 | YYMMDD  HHMM  RPID  LALOLV  QCEVR  CORN  BID  RCPTIM  RPSEC1      |
| NC000002 | WNDSQ1  TMPSQ1  PRSSQ1  <PRSSQ2>  <PRSSQ3>  PCPSQ1  TOCC  HBLCS   |
| NC000002 | {CLDSQ1}  <PPWSQ1>  <RPSEC3>  {RAWRPT}                            |
|----------|-------------------------------------------------------------------|
| NC000007 | YYMMDD  HHMM  {BID}  MTRID  {RCPTIM}  MTRWND  MTRTMP  MTRPRS      |
| NC000007 | MTRVSB  {MTRPRW}  {MTRCLD}  MTRPRC  {RAWRPT}  MTRPKW  SST1  SEST  |
|          |                                                                   |
| NC000100 | SFIDTIME  WIGOSID  RPID  PRESDATA  QMPR  TEMHUMDA  QMAT  QMDD     |
| NC000100 | <BSYEXTM>  VISBSEQN  BSYWND1  QMWN  {BSYWND2}  TPMI  BSYPCP1      |
| NC000100 | {BSYPCP2}  HSALG  HSAWS  GENCLOUD  {BSYSCLD}  {BSYBCLD}           |
| NC000100 | <DIRCLDFT>  VSSO  <PWEATHER>  <STGDSNDM>  GMITH  BID  RCPTIM      |
| NC000100 | CORN  RSRD  EXPRSRD                                               |
|          |                                                                   |
| NC000101 | SFIDTIME  WIGOSID  RPID  PRESDATA  QMPR  TEMHUMDA  QMAT  QMDD     |
| NC000101 | <BSYEXTM>  VISBSEQN  BSYWND1  QMWN  {BSYWND2}  TPMI  BSYPCP1      |
| NC000101 | {BSYPCP2}  HSALG  HSAWS  GENCLOUD  {BSYSCLD}  {BSYBCLD}           |
| NC000101 | <DIRCLDFT>  VSSO  <PWEATHER>  <STGDSNDM>  GMITH  BID  RCPTIM      |
| NC000101 | CORN                                                              |
|          |                                                                   |
| NC000102 | MOBIDENT  WIGOSID  RPID  PRESDATA  QMPR  TEMHUMDA  QMAT  QMDD     |
| NC000102 | <BSYEXTM>  VISBSEQN  BSYWND1  QMWN  {BSYWND2}  TPMI  BSYPCP1      |
| NC000102 | {BSYPCP2}  HSALG  HSAWS  GENCLOUD  {BSYSCLD}  {BSYBCLD}           |
| NC000102 | <DIRCLDFT>  VSSO  <PWEATHER>  <STGDSNDM>  GMITH  BID  RCPTIM      |
| NC000102 | CORN                                                              |
|----------|-------------------------------------------------------------------|
| CLDSQ1   | 361041 | SYNOPTIC REPORT CLOUD DATA                               |
| CLDSQ1   | VSSO  CLAM  CLTP  HOCB  <CLDSQ2>       
| MTRCLD   | 361024 | METAR CLOUD DATA                                         |
| MTRCLD   | VSSO  CLAM  CLTP  HOCB                                            |
|----------|-------------------------------------------------------------------|
| GENCLOUD | 302004 | GENERAL CLOUD INFORMATION                                |
| BSYSCLD  | 361071 | SYNOPTIC REPORT SUPPLEMENTAL CLOUD LAYER DATA (BUFR)     |
| BSYBCLD  | 361072 | SYNOPTIC REPORT CLOUD BASES BELOW STN LEVEL DATA (BUFR)  |
|----------|-------------------------------------------------------------------|
| GENCLOUD | TOCC  VSSO  CLAM  HOCB  CLTP  CLTP  CLTP                          |
| BSYSCLD  | VSSO  CLAM  CLTP  AOFV  HOCB                                      |
| BSYBCLD  | VSSO  CLAM  CLTP  HOCT  CTDS                           
|----------|-------------------------------------------------------------------|
```